File: C#/Aplicatii client-server/labClientServerCSharp/ClientServer/Networking/dto/DTOUtils.cs
using lab3.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Networking.dto
{
    public class DTOUtils
    {
        public static Oficiu getFromDTO(OficiuDTO oficiuDTO)
        {
            string id = oficiuDTO.Id;
            string username = oficiuDTO.Username;
            string password = oficiuDTO.Password;
            return new Oficiu(username, password) { Id = id };
        }

        public static OficiuDTO getDTO(Oficiu oficiu)
        {
            string id = oficiu.Id;
            string username = oficiu.Username;
            string password = oficiu.Password;
            return new OficiuDTO(id, username, password);
        }

        public static Participant getFromDTO(ParticipantDTO participantDTO)
        {
            string id = participantDTO.Id;
            string nume = participantDTO.Nume;
            string echipa = participantDTO.Echipa;
            int capMotor = participantDTO.CapMotor;
            return new Participant(nume, echipa, capMotor) { Id = id };
        }

        public static ParticipantDTO getDTO(Participant participant)
        {
            string id = participant.Id;
            string nume = participant.Nume;
            string echipa = participant.Echipa;
            int capMotor = participant.CapacitateMotor;
            return new ParticipantDTO(id, nume, echipa, capMotor);
        }

        public static Cursa getFromDTO(CursaDTO cursaDTO)
        {
            string id = cursaDTO.Id;
            int nrPersoane = cursaDTO.NrPersoane;
            int capMotor = cursaDTO.CapMotor;
            return new Cursa(nrPersoane, capMotor) { Id = id };
        }

        public static CursaDTO getDTO(Cursa cursa)
        {
            string id = cursa.Id;
            int nrPersoane = cursa.NrPersoane;
            int capMotor = cursa.CapacitateMotor;
            return new CursaDTO(id, nrPersoane, capMotor);
        }

        public static ParticipantCursa getFromDTO(ParticipantCursaDTO participantCursaDTO)
        {
            Tuple<string, string> id = participantCursaDTO.Id;
            string idP = participantCursaDTO.IdParticipant;
            string idC = participantCursaDTO.IdCursa;
            return new ParticipantCursa(idP, idC) { Id = id };
        }

        public static ParticipantCursaDTO getDTO(ParticipantCursa participantCursa)
        {
            Tuple<string, string> id = participantCursa.Id;
            string idP = participantCursa.IdParticipant;
            string idC = participantCursa.IdCursa;
            return new ParticipantCursaDTO(id, idP, idC);
        }

        public static CursaDTO[] getDTO(Cursa[] curse)
        {
            CursaDTO[] curseDTO = new CursaDTO[curse.Length];
            for (int i = 0; i < curse.Length; i++)
            {
                curseDTO[i] = getDTO(curse[i]);
            }
            return curseDTO;
        }

        public static Cursa[] getFromDTO(CursaDTO[] curseDTO)
        {
            Cursa[] curse = new Cursa[curseDTO.Length];
            for (int i = 0; i < curseDTO.Length; i++)
            {
                curse[i] = getFromDTO(curseDTO[i]);
            }
            return curse;
        }

        public static ParticipantDTO[] getDTO(Participant[] participanti)
        {
            ParticipantDTO[] participantiDTO = new ParticipantDTO[participanti.Length];
            for(int i = 0; i < participanti.Length; i++)
            {
                participantiDTO[i] = getDTO(participanti[i]);
            }
            return participantiDTO;
        }

        public static Participant[] getFromDTO(ParticipantDTO[] participantiDTO)
        {
            Participant[] participanti = new Participant[participantiDTO.Length];
            for(int i = 0; i <participantiDTO.Length; i++)
            {
                participanti[i] = getFromDTO(participantiDTO[i]);
            }
            return participanti;
        }
    }
}

File: Java/Aplicatii client-server/labJavaClientServer/Networking/src/main/java/dto/DTOUtils.java
package dto;

import javafx.util.Pair;
import model.Cursa;
import model.Oficiu;
import model.Participant;
import model.ParticipantCursa;

public class DTOUtils {
    public static CursaDTO getDTO(Cursa cursa){
        String id = cursa.getId();
        Integer nrPersoane = cursa.getNrPersoane();
        Integer capMotor = cursa.getCapacitateMotor();
        return new CursaDTO(id, nrPersoane, capMotor);
    }

    public static Cursa getfromDTO(CursaDTO cursaDTO){
        Integer nrPersoane = cursaDTO.getNrPersoane();
        Integer capMotor = cursaDTO.getCapMotor();
        Cursa cursa = new Cursa(nrPersoane, capMotor);
        cursa.setId(cursaDTO.getId());
        return cursa;
    }

    public static Cursa[] getfromDTO(CursaDTO[] curseDTO){
        Cursa[] curse = new Cursa[curseDTO.length];
        for(int i = 0; i < curseDTO.length; i++){
            curse[i] = getfromDTO(curseDTO[i]);
        }
        return curse;
    }

    public static CursaDTO[] getDTO(Cursa[] curse){
        CursaDTO[] curseDTO = new CursaDTO[curse.length];
        for(int i = 0; i < curse.length; i++){
            curseDTO[i] = getDTO(curse[i]);
        }
        return curseDTO;
    }

    public static ParticipantCursaDTO getDTO(ParticipantCursa participantCursa){
        System.out.println("PARTICIPANT CURSA:" + participantCursa);
        Pair<String, String> id = participantCursa.getId();
        String idParticipant = participantCursa.getIdParticipant();
        String idCursa = participantCursa.getIdCursa();
        return new ParticipantCursaDTO(id, idParticipant, idCursa);
    }

    public static ParticipantCursa getFromDTO(ParticipantCursaDTO participantCursaDTO){
        String idParticipant = participantCursaDTO.getIdParticipant();
        String idCursa = participantCursaDTO.getIdCursa();
        ParticipantCursa participantCursa = new ParticipantCursa(idParticipant, idCursa);
        participantCursa.setId(participantCursaDTO.getId());
        return participantCursa;
    }

    public static ParticipantDTO getDTO(Participant participant){
        String id = participant.getId();
        String nume = participant.getNume();
        String echipa = participant.getEchipa();
        Integer capMotor = participant.getCapacitateMotor();
        return new ParticipantDTO(id, nume, echipa, capMotor);
    }

    public static Participant getfromDTO(ParticipantDTO participantDTO){
        String nume = participantDTO.getNume();
        String echipa = participantDTO.getEchipa();
        Integer capMotor = participantDTO.getCapMotor();
        Participant participant = new Participant(nume, echipa, capMotor);
        participant.setId(participantDTO.getId());
        return participant;
    }

    public static OficiuDTO getDTO(Oficiu oficiu){
        String id = oficiu.getId();
        String username = oficiu.getUsername();
        String password = oficiu.getPassword();
        return new OficiuDTO(id, username, password);
    }

    public static Oficiu getfromDTO(OficiuDTO oficiuDTO){
        String username = oficiuDTO.getUsername();
        String password = oficiuDTO.getPassword();
        Oficiu oficiu = new Oficiu(username, password);
        oficiu.setId(oficiuDTO.getId());
        return oficiu;
    }
}
